### ISSUE TITLE
Initial Review

### ISSUE THREAD

**Issue by dschier-wtd** [while-true-do/ansible-role-postfix#1]
### Type of Report

Review Request

### Actual behaviour

Role had no initial review.

### Expected behaviour

Role must be reviewed and commented.

**Comment by daniel-schier** [comment]
Initial review done from my side.
@donvipre or @Cinux90 would be awesome, if you can have another look now.

**Comment by Cinux90** [comment]
Points from my side:
- empty folders can be removed:
  - vars
  - files
- .git_keep file can be removed in folder templates
- Why we need to restart the service via handler, if the last task start the service too? Are there any reason for that? If not i think it can be combined.
- template/main.cf.j2:
  - Yesterday I learned that an if expression in j2 should be looks like:
`{% if <variablenname> is defined %} ` is to check if the variable is defined. In Line [76](https://github.com/while-true-do/ansible-role-postfix/blob/master/templates/main.cf.j2#L76), [88](https://github.com/while-true-do/ansible-role-postfix/blob/master/templates/main.cf.j2#L88), [107](https://github.com/while-true-do/ansible-role-postfix/blob/master/templates/main.cf.j2#L107), [326](https://github.com/while-true-do/ansible-role-postfix/blob/master/templates/main.cf.j2#L326), [434](https://github.com/while-true-do/ansible-role-postfix/blob/master/templates/main.cf.j2#L434) you check if the varibale is True. 
  - Furthermore you check by line 88 for **wtd_postfix_myhostname** but set variable **wtd_postfix_mydomain** at the line below. Looks for me like a typo.
  - In Line [78](https://github.com/while-true-do/ansible-role-postfix/blob/master/templates/main.cf.j2#L78),[91](https://github.com/while-true-do/ansible-role-postfix/blob/master/templates/main.cf.j2#L91),[99](https://github.com/while-true-do/ansible-role-postfix/blob/master/templates/main.cf.j2#L90),[109](https://github.com/while-true-do/ansible-role-postfix/blob/master/templates/main.cf.j2#L109),434 you have an else expression but dont add any content, which not start with _#_, into the template. I'm not aware with postfix configs at all, but tthis variables looks mandatory.

**Comment by daniel-schier** [comment]
@Cinux90 
- I'll remove the files and directories
- restart ensures that the service is restarted. the started only ensures that it is running. if the service was already running, it wont restart it. so you have to have a handler for this.
- template/main.cf.js2
  - I dont want to check for definition, but if it contains something
  - will fix this
  - i have the default comments in it, which should be present only if nothing is provided

**Comment by daniel-schier** [comment]
@Cinux90 done

Note to the check for definition / content:
- checking for definitions should be done ONLY, if a variable can be absent.
- having the variable in defaults with "empty content" should be used, if possible to ensure it is documented and easy to read

tl;dr: 
- If the variable must be undefined to be used properly -> use "if defined"
- If the variable can be empty to have the proper check done -> use "if variable"

**Comment by Cinux90** [comment]
@daniel-wtd Thanks for the information.
For my understanding `wtd_postfix_hostname: ""` results into True in an if statement. But thats not the case. :)
Rest looks ok.

**Comment by daniel-schier** [comment]
Seems like everything is ok for now
